dev/bin/interaction/delete-test-resources.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+source "$(dirname "${BASH_SOURCE[0]}")/common.sh"
+
+# Script used to delete Bridges and Processors used in E2E tests
+# Script will delete Bridges without Processors, for Bridges with Processors it will delete Processors if they are in proper state
+# Test Bridges are identified using "test-" prefix in their names
+
+BRIDGES=($(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/v1/bridges | jq '.items[] | select(.name | startswith("test-")) | .id' | tr -d \"))
+
+for i in "${BRIDGES[@]}"
+do
+   NUMBER_OF_PROCESSORS=$(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/v1/bridges/$i/processors | jq '.items | length')
+  if [ "$NUMBER_OF_PROCESSORS" == "0" ]; then
+    echo "No processors in Bridge $i."
+    BRIDGE_STATUS=$(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/v1/bridges/$i | jq '.status' | tr -d \")
+    if [ "$BRIDGE_STATUS" == "ready" ] || [ "$BRIDGE_STATUS" == "failed" ]; then
+      echo "Deleting Bridge $i"
+      curl -s -H "Authorization: $OB_TOKEN" -X DELETE $MANAGER_URL/api/v1/bridges/$i
+    else
+      echo "Bridge $i status $BRIDGE_STATUS is incompatible with automated deletion, please remove it manually"
+    fi
+  else
+    echo "$NUMBER_OF_PROCESSORS processors found in Bridge $i"
+    PROCESSORS=($(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/v1/bridges/$i/processors | jq '.items[].id' | tr -d \"))
+
+    for j in "${PROCESSORS[@]}"
+    do
+      PROCESSOR_STATUS=$(curl -s -H "Authorization: $OB_TOKEN" -X GET $MANAGER_URL/api/v1/bridges/$i/processors/$j | jq '.status' | tr -d \")
+      if [ "$PROCESSOR_STATUS" == "ready" ] || [ "$PROCESSOR_STATUS" == "failed" ]; then
+        echo "Deleting Processor $j"
+        curl -s -H "Authorization: $OB_TOKEN" -X DELETE $MANAGER_URL/api/v1/bridges/$i/processors/$j
+      else
+        echo "Processor $j status $PROCESSOR_STATUS is incompatible with automated deletion, please remove it manually"
+      fi
+    done
+  fi
+done
```

Comment:
In the case you have processors for the bridge you are not deleting the bridge itself then

Reply:
Yes, because right now it is not possible to delete Bridges which contains Processes.
So the possibility was either to implement some sort of waiting loop - that would bring some complexity..... or leave it like this so user should rerun the script several times. So when there is a Bridge with Processors then first run will delete Processors, user should wait some time and then rerun the script again to remove Bridges.

Reply:
Could you add a comment about this? So that the expectation is that this script is not going to delete all the resources at the first execution but it has to be run multiple times

Reply:
Added a comment, added an echo mentioning a need to rerun the script if the Bridge has some Processors